Automattic/jetpack
# JS Components: Convert JetpackLogo component to TypeScript

#### Changes proposed in this Pull Request:
* Convert `JetpackLogo` component to TypeScript and cleans up the propTypes.
* Convert the `JetpackLogo` stories to TS
* Convert the `JetpackLogo` unit tests to TS
* Get rid of `enzyme` in unit test and rather use `@testing-library/react`

Note: Removing enzyme is necessary in order to convert unit tests to TS.

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

* Checkout the PR
* Run `cd projects/js-packages/storybook && pnpm run storybook:dev`
* Confirm that all the stories of `JetpackLogo` component work as expected
* Run `jetpack test js-packages/components js`
* Confirm that all the tests pass
* Optionally Check the consumers of `JetpackLogo` and related components and confirm that they work as expected
* Check for any regression

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - 0-as-1202276314264002
